app/Http/Controllers/MigrateController.php

```
@@ -12,9 +12,34 @@ class MigrateController extends Controller
     public function migrate(Request $request, string $openlistId)
     {
         // The "legacy-" prefix protects against high-jacking from another GUID.
-        $materials = DB::table('materials')
-            ->where(['guid' => 'legacy-' . $openlistId])
-            ->update(['guid' => $request->user()->getId()]);
+        // We have to use a raw query here as Illuminate/DB does not support
```

I'd just use a try/catch and ignore errors. I guess if you'd like to differentiate duplicate key from other errors, you could check that SQLSTATE is 23xxx.

---

@xendk Could you give an example?

As I see it the UPDATE query either works on all the rows or none of them, so ignoring the error means that users with collisions will never have any of their materials migrated.

If we want to ignore errros we have to fetch all materials and update them individually.

---

I'll just keep this as a single query for now.